### PR TITLE
US-887: Create AccountNumber component

### PR DESCRIPTION
**Pull request by anoveskey**

### CODE REVIEW

**Comment by anoveskey** [comment]
@astoneDTE I will double check it to make sure something isn't missing and let you know in chat. Hopefully I didn't forget to include a file or something...

**Comment by anoveskey** [comment]
> @astoneDTE I will double check it to make sure something isn't missing and let you know in chat. Hopefully I didn't forget to include a file or something...

Turns out I forgot to run the update on the App component snapshot...

File: app/components/AccountNumber/index.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+
+import './index.less';
+
+interface IAccountNumber {
```
Comment (jmiller-nexient) [inline app/components/AccountNumber/index.tsx:5]: Always include the suffix, "Props" on interfaces like this.  IAccountNumber would represent an entity not the props passed into a component.

File: app/components/AccountNumber/index.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  value: string;
+}
+
+const AccountNumber: React.FC<IAccountNumber> = (props: IAccountNumber) => {
+  const { value } = props;
+  return (
```
Comment (jmiller-nexient) [inline app/components/AccountNumber/index.tsx:11]: Per clean code practices, always include a blank line before all return statements (unless it is the only statement).  See the bottom of this page: https://nexient.atlassian.net/wiki/spaces/DTE/pages/1040581086/MIMO+React+Coding+Guidelines

File: app/components/AccountNumber/index.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    <span className="dte-wismo-account-number">
+      {value}
```
Comment (jmiller-nexient) [inline app/components/AccountNumber/index.tsx:13]: I took a look at my account online and the wireframes don't match the way I see the account number formatted this way:
image

Comment (anoveskey) [reply app/components/AccountNumber/index.tsx:13]: I'm not sure what the regex for this should look like, the numbers seem to be grouped: 

4-3-4-1 

Is that a consistent pattern that we should follow? Do we know that all account numbers will be the same length of numbers?

Comment (jmiller-nexient) [reply app/components/AccountNumber/index.tsx:13]: Yes, let's make that assumption for now.

File: app/components/AccountNumber/tests/index.test.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React, { ReactElement } from 'react';
+import renderer from 'react-test-renderer';
+
+import AccountNumber from '..';
+
+describe('<AccountNumber />', () => {
+  it('should render correctly', () => {
+    const tree: ReactElement = renderer.create(
+      <AccountNumber value={'123456'} />,
```
Comment (jmiller-nexient) [inline app/components/AccountNumber/tests/index.test.tsx:9]: It's interesting to me that this test didn't fail when there are less than 12 digits.  I think we should have a guard statement in there to throw an error if the value has less than or more than 12 digits.  Then add two tests for those cases.

Comment (anoveskey) [reply app/components/AccountNumber/tests/index.test.tsx:9]: And how should we handle that exactly? If the number passed in is shorter we pad it with zeros first, or?

Comment (jmiller-nexient) [reply app/components/AccountNumber/tests/index.test.tsx:9]: I confirmed that account number is always exactly 12 numeric digits.  So, if `value` has more or less than 12 digits, throw a new error that says something like 'account numbers must be 12 numeric digits.'

**Pull request merged**